### PR TITLE
feat: populate rules status

### PR DESCRIPTION
**Pull request by TheSpiritXIII**
Add MonitoringStatus to all Rules CRDs.

### CODE REVIEW

File: pkg/operator/apis/monitoring/v1/rules_types.go [hunk 1]
```diff
@@ -192,5 +204,5 @@ type Rule struct {
 
 // RulesStatus contains status information for a Rules resource.
 type RulesStatus struct {
-	// TODO: add status information.
+	MonitoringStatus `json:",inline"`
```
Comment (maxamins) [inline pkg/operator/apis/monitoring/v1/rules_types.go:207]: I don't see in the CRD. Do you need to run `make regen`?

File: charts/operator/crds/monitoring.googleapis.com_clusterrules.yaml [hunk 2]
```diff
@@ -124,6 +124,44 @@ spec:
             type: object
           status:
             description: Most recently observed status of the resource.
+            properties:
```
Comment (maxamins) [inline charts/operator/crds/monitoring.googleapis.com_clusterrules.yaml:127]: What caused this piece to be generated? I don't see a matching change.

File: pkg/operator/apis/monitoring/v1/monitoring_types.go [hunk 3]
```diff
@@ -77,17 +81,17 @@ type MonitoringStatus struct {
...
 	var (
 		specChanged              = status.ObservedGeneration != gen
 		statusTransition, update bool
 		conds                    = make(map[MonitoringConditionType]*MonitoringCondition)
 	)
 
-	if cond.Type == "" || cond.Status == "" {
-		return update, errInvalidCond
+	if !cond.IsValid() {
+		return false
```
Comment (maxamins) [inline pkg/operator/apis/monitoring/v1/monitoring_types.go:94]: How often does this happen? Should we log when this happens?

**Review by maxamins — COMMENTED** [review]
Nice job! few comments.

File: charts/operator/crds/monitoring.googleapis.com_clusterrules.yaml (hunk 2 above)
Comment (TheSpiritXIII) [reply charts/operator/crds/monitoring.googleapis.com_clusterrules.yaml:127]: This is the addition of the `MonitoringCondition` field.

File: pkg/operator/apis/monitoring/v1/monitoring_types.go (hunk 3 above)
Comment (TheSpiritXIII) [reply pkg/operator/apis/monitoring/v1/monitoring_types.go:94]: In practice, never. We always call `SetMonitoringCondition` with a valid condition.

If someone wants to build a `MonitoringCondition` dynamically and wants logging, they could call `IsValid` and then log.

File: pkg/operator/apis/monitoring/v1/rules_types.go (hunk 1 above)
Comment (TheSpiritXIII) [reply pkg/operator/apis/monitoring/v1/rules_types.go:207]: It's in there (the `properties` field). Correct me if I'm wrong.

File: charts/operator/crds/monitoring.googleapis.com_clusterrules.yaml (hunk 2 above)
Comment (maxamins) [reply charts/operator/crds/monitoring.googleapis.com_clusterrules.yaml:127]: `MonitoringCondition` seems to have been added in a prior release. So it's strange that this is showing up now.

File: pkg/operator/apis/monitoring/v1/rules_types.go (hunk 1 above)
Comment (maxamins) [reply pkg/operator/apis/monitoring/v1/rules_types.go:207]: I see it. Thanks!

File: pkg/operator/apis/monitoring/v1/monitoring_types.go (hunk 3 above)
Comment (maxamins) [reply pkg/operator/apis/monitoring/v1/monitoring_types.go:94]: OK!

**Review by maxamins — APPROVED** [review]

File: charts/operator/crds/monitoring.googleapis.com_clusterrules.yaml (hunk 2 above)
Comment (TheSpiritXIII) [reply charts/operator/crds/monitoring.googleapis.com_clusterrules.yaml:127]: The definition is copied into each place it's used. This is the first time it's used in `ClusterRules`!

**Pull request merged**